db_test.go

```diff
@@ -1337,3 +1338,50 @@ func TestCorrectNumTombstones(t *testing.T) {
 	testutil.Ok(t, db.Delete(9, 11, labels.NewEqualMatcher("foo", "bar")))
 	testutil.Equals(t, uint64(3), db.blocks[0].meta.Stats.NumTombstones)
 }
+
+// TestDisableCompactions checks that we can disable the
+// auto compaction and run compactions on demand.
+// This is needed for unit tests that rely on
+// checking state before and after a compaction.
+func TestDisableAutoCompactions(t *testing.T) {
+	db, close := openTestDB(t, nil)
+	defer close()
+	defer db.Close()
+
+	db.DisableCompactions()
```

Comment:
For more comprehensive test, should we also check
* Auto compaction is happening before disabling.
* Enable auto compactions again at the end and check if auto compactions are happening.

(Enabled -> Disabled -> Enabled)

Reply:
I refactored it a bit to better cover `Disabled -> Enabled` and unless you can think of some immediate bug that that is not covered by this logic. I think it should be ok.